lib/Epub/Epub/parsers/ChapterHtmlSlimParser.cpp

```diff
@@ -249,6 +254,16 @@ bool ChapterHtmlSlimParser::parseAndBuildPages() {
       return false;
     }
 
+    // Update progress (call every 10% change to avoid too frequent updates)
+    bytesRead += len;
+    if (progressFn && totalSize > 0) {
+      const int progress = static_cast<int>((bytesRead * 100) / totalSize);
+      if (progress != lastProgress && progress % 10 == 0) {
+        lastProgress = progress;
+        progressFn(progress);
+      }
```

Comment:
progress has to fall on a 10% exactly to get an update, instead it'd be better to do something like: `lastProgress / 10 != progress / 10`